pkg/token/daemon.go

```diff
@@ -143,26 +143,94 @@ func newDaemon(idConfig *config.IdentityConfig, tt mode) (*daemon, error) {
 	}, nil
 }
 
-func (d *daemon) updateTokenWithRetry() error {
+func (d *daemon) updateTokenCaches() <-chan error {
+	atTargets := d.accessTokenCache.Keys()
+	rtTargets := d.roleTokenCache.Keys()
+	log.Infof("Attempting to fetch tokens from Athenz ZTS server: access token targets[%v], role token targets[%v]...", atTargets, rtTargets)
+	echan := make(chan error, len(atTargets)+len(rtTargets))
+	go func() {
+		defer close(echan)
+		atErrorCount := 0
+		for _, t := range atTargets {
+			key := t // prevent closure over loop variable
+			err := d.updateTokenWithRetry(key, mACCESS_TOKEN)
+			if err != nil {
+				echan <- err
+				atErrorCount += 1
+			}
+		}
```

Comment:
```suggestion
		var wg sync.WaitGroup
		for _, t := range atTargets {
		    wg.Add(1)
		    go func(key string) {
                        defer wg.Done()
                        err := d.updateTokenWithRetry(key, mACCESS_TOKEN)
			if err != nil {
				echan <- err
				atErrorCount += 1 // change to atomic.Int
			}
                   }(t)
		}
```
client sidecar do parallel based on key, 1 key = 1 routine.
please use https://gobyexample.com/waitgroups to wait for all routines to finish.

Reply:
fixed: https://github.com/AthenZ/k8s-athenz-sia/pull/75/commits/471fb65a87c711df633ef710677f9990eceb26fe